python/cuml/cuml/naive_bayes/naive_bayes.py

```diff
@@ -409,6 +410,12 @@ def _partial_fit(
             check_dtype=expected_y_dtype,
         ).array
 
+        if X.shape[0] != y.shape[0]:
+            raise ValueError(
+                "X and y must have the same number of samples. "
+                f"Got {X.shape[0]} and {y.shape[0]}."
+            )
+
```

Comment:
The idiomatic way in cuml to handle this would be to pass `check_cols=X.shape[0]` to `input_to_cupy_array` above.